### ISSUE TITLE
SIGSEGV when first refresh operation fails

### ISSUE THREAD

**Issue by kadodson612** [GoogleCloudPlatform/cloudsql-proxy#745]
## Bug Description

CloudSQL Proxy throws SIGSEGV once the first MySQL client connection is made and the proxy is started with "-ip_address_types=PRIVATE" and the CloudSQL instance does not have a private IP.  

```
./cloud_sql_proxy -instances=PROJECT_ID:us-central1:XYZ=tcp:3306 -verbose -ip_address_types=PRIVATE

2021/04/20 12:18:11 Rlimits for file descriptors set to {Current = 8500, Max = 9223372036854775807}
2021/04/20 12:18:11 using credential file for authentication; path="/Users/……./cred.json"
2021/04/20 12:18:12 Listening on 127.0.0.1:3306 for PROJECT_ID:us-central1:XYZ
2021/04/20 12:18:12 Ready for new connections
2021/04/20 12:18:22 New connection for "PROJECT_ID:us-central1:XYZ"
2021/04/20 12:18:22 refreshing ephemeral certificate for instance PROJECT_ID:us-central1:XYZ
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x155a56f]

goroutine 83 [running]:
github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy.(*Client).startRefresh.func1(0xc00032a540, 0xc0002e6180, 0x7ffeefbff5ad, 0x48, 0x45d964b800)
	/workspace/proxy/proxy/client.go:285 +0x38f
created by github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy.(*Client).startRefresh
	/workspace/proxy/proxy/client.go:264 +0x85


(venv) [user@machine:~/software] $ cloud_sql_proxy -version
Cloud SQL Auth proxy: 1.21.0+darwin.amd64
```

## How to reproduce (see above)

  1. Download and run proxy version 1.21.0+darwin.amd64
  2. Try to connect to DB without private IP via proxy (using private IP flag), resulting in signal SIGSEGV: segmentation violation

## Environment

1. OS type and version: <YOUR_OS_HERE>
2. Cloud SQL Proxy version (`./cloud_sql_proxy -version`): Cloud SQL Auth proxy: 1.21.0+darwin.amd64

**Comment by kurtisvg** [comment]
It looks like what's happening is the initial refresh fails, so the "old" `cfg` field is used, which is nil because this is the first connection attempt: 

https://github.com/GoogleCloudPlatform/cloudsql-proxy/blob/9347d99c08167f80c1671ee6d57e0dec1276622f/proxy/proxy/client.go#L266-L285

**Comment by enocom** [comment]
A fix for this is available in [v1.22.0](https://github.com/GoogleCloudPlatform/cloudsql-proxy/releases/tag/v1.22.0).
